rtf_vis_tool/src/Components/Bundle_Adj_PC.js

```diff
@@ -0,0 +1,125 @@
+                var finalPointsJSX = [];
+                
+                // Remove the first 3 commented lines of points3D.txt.
+                for (var i = 0; i < 3; i++) {
+                    arrStringPoints.shift();
+                }
+
+                const arrNumPoints = arrStringPoints.map(point => point.split(" ").map(Number));
+                setPointCloudRaw(arrNumPoints);
+
+                // Loop through array. convert strings to numbers. Append to final point cloud.
+                for (var index = 0; index < arrNumPoints.length; index += 1) {
+                    var pointArr = arrNumPoints[index];
+                    
+                    finalPointsJSX.push(
+                        <PointMesh  
+                            position={[pointArr[1], pointArr[2], pointArr[3]]}  
+                            color={`rgb(${pointArr[4]}, ${pointArr[5]}, ${pointArr[6]})`} 
+                            size={pointSizeArr}/>
+                    );
+                }
+                setPointCloudJSX(finalPointsJSX);
+            })
+    }, []);
+
+    /* Function that updates the radius of all points within a point cloud.
+       Called everytime the react slider input is interacted with.
+    */
+    const updatePointSizes = (radius) => {
+        var finalPointsJSX = [];
+        for (var i = 0; i < pointCloudRaw.length; i += 1) {
+            var pointArr = pointCloudRaw[i];
+            
+            finalPointsJSX.push(
+                <PointMesh  
+                    position={[pointArr[1], pointArr[2], pointArr[3]]}  
+                    color={`rgb(${pointArr[4]}, ${pointArr[5]}, ${pointArr[6]})`} 
+                    size={[radius]}/>
+            );
+        }
+        setPointCloudJSX(finalPointsJSX);
+    }
+
+    return (
+        <div className="ba-container">
+            <h2>Bundle Adjustment Point Cloud</h2>
+            <Canvas colorManagement camera={{ fov: 20, position: [50, 50, 50], up: [0,0,1]}}>
+                <ambientLight intensity={0.5}/>
+                <pointLight position={[100, 100, 100]} intensity={1} castShadow />
+                <pointLight position={[-100, -100, -100]} intensity={0.8}/>
+                <directionalLight 
+                    position={[0,20,0]} 
+                    intensity={1.5} 
+                    shadow-mapSize-width={1024}
+                    shadow-mapSize-height={1024}
+                    shadow-camera-far={50}
+                />
+        
+                {pointCloudJSX}
+                {showCoordGrid && <CoordinateGrid />}
+                <OrbitControlsComponent />
+            </Canvas>
+
+            <button className="ba_go_back_btn" onClick={() => props.toggleDA_PC(false)}>Go Back</button>
+            <button className="toggle_grid_btn" onClick={() => setShowCoordGrid(!showCoordGrid)}>
+                Toggle Coordinate Grid
+            </button>
+
+            <div className="point_size_slider">
```

Comment:
Consider making into a separate component. In general , no function/component should be longer than 4 lines, give or take 5 ;-)